nixos/modules/services/networking/bee-clef.nix

```diff
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.bee-clef;
+in {
+  meta = {
+    maintainers = with maintainers; [ attila-lendvai ];
+  };
+
+  ### interface
+
+  options = {
+    services.bee-clef = {
+      enable = mkEnableOption "clef external signer instance for Ethereum Swarm Bee";
+
+      data-dir = mkOption {
+        type = types.nullOr types.str;
+        default = "/var/lib/bee-clef";
+        description = "Data dir for bee-clef.";
+      };
+
+      password-file = mkOption {
+        type = types.nullOr types.str;
+        default = "/var/lib/bee-clef/password";
+        description = "Password file for bee-clef.";
+      };
+    };
+  };
+
+  ### implementation
+
+  config = mkIf cfg.enable {
+    systemd.packages = [ pkgs.bee-clef ];
+    # if we ever want to have rules.js under /etc/bee-clef/
+    # environment.etc."bee-clef/rules.js".source = ${pkgs.bee-clef}/rules.js
+
+    services.bee-clef = {
+      data-dir      = "/var/lib/bee-clef";
+      password-file = "/var/lib/bee-clef/password";
+    };
+
+    systemd.tmpfiles.rules = [
+        "d '${cfg.data-dir}' 0750 bee-clef bee-clef"
+        "d '${cfg.data-dir}/keystore' 0700 bee-clef bee-clef"
+      ];
+
+    systemd.services.bee-clef = {
+      wantedBy = [ "bee.service" ];
+      serviceConfig = {
+        ExecStart = [
+          "" # this hides/overrides what's in the original entry
+          "${pkgs.bee-clef}/share/bee-clef/clef-service start"
+        ];
+        ExecStop = [
+          "" # this hides/overrides what's in the original entry
+          "${pkgs.bee-clef}/share/bee-clef/clef-service stop"
+        ];
+        # NOTE this would create the dir with the wrong permissions
+        #StateDirectory = "${cfg.data-dir}";
+      };
+
+      # NOTE for now nothing is installed into /etc/bee-clef/. the config files are used as read-only from the package.
+      preStart = with cfg; ''
```

Comment:
Personally I find the `with cfg;` bit confusing... you don't gain much, I think it might be best to leave as `${cfg.data-dir}`... but that is just my opinion - feel free to entirely disregard this comment :+1: